### ISSUE TITLE
increase timeout for app removal to 24h

### ISSUE THREAD

**Issue by AquariusPower** [raboof/nethogs#153]
I use mobile internet limited to 100MB/day... :(
I need to know exactly what apps are abusing of the data usage unnecessarily like:
- software update (I let it start w/o internet and it stay put w/o trying to use data apparently)
- dropbox that to upload 1MB uploads 5+MB :( non-sense...
- chrome uses much more data than konqueror
- (this list never ends..)
well, at least that is what I was able to read on nethogs.

now, I would like to let it measuring each app usage for a whole day.
I tried increasing timeout to 3 days, but it crashed on 46min, so I limited to 45min for app removal from the list.

my question is: any tips? is it doable or I will have to change too much?

- I would like to add a command line to set the timeout.
- also on the list, to add a column saying for how long the app is using the internet (as YYYY/MM/DD-H:M:S for a start time, is quite bigger than DDD-H:M:S)

**Comment by 4goodapp** [comment]
> now, I would like to let it measuring each app usage for a whole day.
Dis you try using NetHogs with the folowing command:
`sudo nethogs -v 3` ?

**Comment by AquariusPower** [comment]
@4goodapp yep, but that is not related to timeout,
in a sense that even with that (that I use all times), the timeout will still happen.

Anyway, my "timeout tweak only" compilation from this branch was crashing,
I wonder if there could have some stack trace enabled so I could drop it here?

even with 45min only of timeout it crashed, so I had to use the one installed on the system again.

any tips?

**Comment by raboof** [comment]
You could attach [gdb](https://www.gnu.org/software/gdb/) to the program to view its state when it crashes. Would be interesting!

**Comment by choekstr** [comment]
This would be incredibly useful as of right now if you aren't actively watching it you miss most of the things that are using bandwidth.  This is good for an instantaneous read but with a simple adjustment to the timeout value (when items are removed from the list) we could have a longer term monitoring and analysis tool that would help everyone with bandwidth constraints.

**Comment by raboof** [comment]
While 'live' usage is nethogs' main use case, I'd be happy to review a PR that makes the timeout length configurable.

**Comment by AquariusPower** [comment]
despite I like the idea of gdb, I never used it actually :>
I would like to plug my [dbg](https://github.com/AquariusPower/CppDebugMessages/blob/master/dbgmsg.cpp) code (is not a  lib yet..) on it for tests but hadnt yet,
or at least this part:
```
  signal(SIGHUP , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGINT , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGQUIT, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGILL , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGABRT, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGFPE , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGKILL, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGSEGV, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGPIPE, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGTERM, dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGBUS , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGSYS , dbgmsg::SigHndlr);DBGLNSELF;
  signal(SIGTRAP, dbgmsg::SigHndlr);DBGLNSELF;
```

**Comment by choekstr** [comment]
After reading some of the other issues it was referenced that the later versions had a longer timeout so I checked out and compiled from git and it is exactly as I would like: a persistent list of processes and the send/received amount for a long period of time.  

Might be time to release a new version since the Ubuntu repos have 0.8.1 still...

**Comment by raboof** [comment]
@choekstr feel free to lobby! Debian testing and unstable are at 0.8.5, and I'd happily encourage distro's to just ship 'master'

**Comment by AquariusPower** [comment]
@raboof ugh, it was 2 months ago, I should update thx!! :)

**Comment by AquariusPower** [comment]
It is working really great now!! thx!!

btw, I use this on ubuntu that may help others too (after locally cloning this repo):
```
make && sudo -k checkinstall --pkgsource="https://github.com/raboof/nethogs/" --pkglicense="GPLv2" --deldesc=no --nodoc --maintainer="$USER\\<$USER@$HOSTNAME\\>" --pkgarch=$(dpkg --print-architecture) --pkgversion="0.8.5" --pkgrelease="MakeRelease" --pkgname=nethogs make install # see doc/RELEASING.md for the version and release fields, obs the release cant have spaces.
```

**Comment by AquariusPower** [comment]
I am using: NetHogs version 0.8.5-45-g2e8a4f8 (cloned a few days ago here)
If I run it before connecting to my USB tethering, it will not show new connections after the connection is established. 
I run it like `nethogs -d 1 -c 0 -v 3 -a`

Another thing that happens is that it crashes if for ex. I open chromium and instantly closes it. 
When I turn on airplane mode and off again, it may close too.

I learned how to use ddd or nemiver, the only problem I have is how to let the application started thru them to automatically run, I always have to click a button and that is quite annoying... I will try nethogs thru them once today see if I get something useful (like a stack).

should I open a new issue?

**Comment by raboof** [comment]
@AquariusPower yes, these issues sound like they're different from increasing the timeout for app removal to 24h, so please create separate issues :)
